setup.sh

```diff
@@ -15,6 +18,19 @@ sudo apt-get install -y postgresql postgresql-contrib
 echo "install requirements"
 pip3 install -r /vagrant/requirements.txt
 
-echo "running flask_init.py"
-export FLASK_APP=/vagrant/flask_init.py
-python3 -m flask run --host=0.0.0.0 >> /vagrant/log.log 2>&1 &
+echo "configuring database"
```

Comment:
Executing this kind of SQL manually during the setup stage means you get a new database every time you bring up the system - this might work out in development but not in staging or production.

The technique for managing database configuration over time with Git is called "database migrations" here is a post introducing this concept:
https://rollout.io/blog/database-migration/
The examples there are for Ruby-on-Rails but the same concepts apply to Flask and SQLAlchemy

Here is a DB migration framework you can use with SQLAlchemy:
https://alembic.sqlalchemy.org/en/latest/

Reply:
> Executing this kind of SQL manually during the setup stage means you get a new database every time you bring up the system

Are you sure ? When I tried to create a database twice I got an error.

> The technique for managing database configuration over time with Git is called "database migrations"

There is a flask extension called Flask-Migrate that handles SQLAlchemy database migrations for Flask applications using Alembic. It super easy to use and it configures Alembic in the proper way to work with our Flask and SQLAlchemy application.
So I'll add that to our project.

Regarding the initial creation of the database, what is the proper way doing so ?
Should we create the database manually ? But on the next vagrant up we'll need to create it again...
That is why I added those lines to setup.sh so we wouldn't need to do it over and over again.

Reply:
> Executing this kind of SQL manually during the setup stage means you get a new database every time you bring up the system

Should we create the database manually ? But on the next vagrant up we'll need to create it again... That is why I added those lines to setup.sh so we wouldn't need to do it over and over again.

> The technique for managing database configuration over time with Git is called "database migrations"

Doesn't it migrate only the tables schema without the data ? and if so on vagrant destroy all the data will be gone. How does it different from creating a new database on every vagrant up ?

Reply:
I squashed some commits so this file is no longer reachable through this commit.
This file could be found [here](https://github.com/knowabli/edison/blob/61e4b1c49d53c62a100759c7543f3c03c7e1af3e/setup.sh).